### PR TITLE
[DOCS] Document `time_series_metric` mapping parameter

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the `time_series_metric` mapping parameter added with PR #76766.

Also renames the `dimension` parameter to `time_series_dimension` for PR #78012.

### Previews
- `aggregate_metric_double`: https://elasticsearch_78013.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/aggregate-metric-double.html#aggregate-metric-double-params
- `histogram`: https://elasticsearch_78013.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/histogram.html#histogram-params
- Numeric field types: https://elasticsearch_78013.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/number.html#number-params

- `ip`: https://elasticsearch_78013.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ip.html#ip-params
- `keyword`: https://elasticsearch_78013.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/keyword.html#keyword-params
- Index settings: https://elasticsearch_78013.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/index-modules.html#index-mapping-dimension-fields-limit

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/mapping/types/numeric.asciidoc [hunk 1]
```diff
@@ -173,9 +175,16 @@ support this parameter.
...
+// tag::time_series_metric[]
+`time_series_metric`::
+For internal use by Elastic only. Marks the field as a time series metric. The
+value is the metric type. Defaults to `null` (Not a time series metric).
+// end::time_series_metric[]
++
+For numeric fields, this parameter accepts `gauge` and `counter`. You can't
+update this parameter for existing fields.
++
+`scaled_float` and `unsigned_long` fields don't support this parameter.
```
Comment (csoulios) [inline docs/reference/mapping/types/numeric.asciidoc:187]: I am not sure where does the constraint that `scaled_float` and `unsigned_long` fields don't support this parameter. There is no such constraint in the code. Am I missing something here?

**Review by csoulios — COMMENTED** [review]
I left one comment for numeric fields. 

Also, in PR #78012 I renamed `dimension` parameter to `time_series_parameter`. Not sure if you want to add this change in the same or a separate PR.

File: docs/reference/mapping/types/numeric.asciidoc (hunk 1 above)
Comment (csoulios) [reply docs/reference/mapping/types/numeric.asciidoc]: Thank you for opening the issue.  I will check that asap.

File: docs/reference/mapping/types/numeric.asciidoc [hunk 2]
```diff
@@ -173,9 +170,20 @@ support this parameter.
     the <<mapping-source-field,`_source`>> field. Accepts `true` or `false`
     (default).
 
-<<mapping-field-meta,`meta`>>::
+include::keyword.asciidoc[tag=dimension]
++
+Of the numeric field types, only `byte`, `short`, `integer`, and `long` fields
+support this parameter.
```
Comment (csoulios) [inline docs/reference/mapping/types/numeric.asciidoc:176]: `unsigned_long` is also supported. Added in #78204

File: docs/reference/mapping/types/numeric.asciidoc [hunk 3]
```diff
@@ -173,9 +170,20 @@ support this parameter.
...
-    Metadata about the field.
+// tag::time_series_metric[]
+`time_series_metric`::
+(Optional, string) For internal use by Elastic only. Marks the field as a time
+series metric. The value is the metric type. Defaults to `null` (Not a time
+series metric).
+// end::time_series_metric[]
++
+For numeric fields, this parameter accepts `gauge` and `counter`. You can't
+update this parameter for existing fields.
```
Comment (csoulios) [inline docs/reference/mapping/types/numeric.asciidoc:186]: I think we should also mention that for all numeric metrics, `doc_values` parameters must be `true`

**Review by csoulios — COMMENTED** [review]
Changes look good. I left some more comments.

**Comment by csoulios** [comment]
One last thing we should add is that a numeric field cannot be both a dimension and a metric.
I have added this constraint in #78204

**Comment by jrodewig** [comment]
Thanks @csoulios. I documented the constraint with 8354931550b00ac97e855a453ae84bade809f78f.
I also noted that `unsigned_long` fields can be a time series dimension.

**Review by csoulios — APPROVED** [review]
LGTM. Thank you for iterating so fast on this!

**Pull request merged**